src/box/gc.c

```diff
@@ -735,6 +809,238 @@ gc_consumer_name(struct gc_consumer *consumer)
 			  tt_uuid_str(&consumer->uuid));
 }
 
+/**
+ * Write the consumer to persistent WAL GC state.
+ */
+static int
+gc_consumer_persist(struct gc_consumer *consumer)
+{
+	if (!gc_has_persistent_consumers())
+		return 0;
```

Comment:
I think this should be checked before this function is called. For example, there's no need to begin a transaction in `gc_consumer_sync` if the space doesn't exist.

Reply:
Done, replaced this condition with assertion.